File: constantoutput.m
clc;
clear all;
close all;

image = imread('lena.bmp');

prompt = 'Enter the first threshold : ';
lower_threshold = input(prompt);
prompt = 'Enter the 2nd threshold : ';
higher_threshold = input(prompt);
prompt = 'Enter the constant for above thresholds : ';
constant = input(prompt);

[r,c] = size(image);
img = image;

for i=1:1:r
    for j=1:1:c
        if(img(i,j) > lower_threshold && img(i,j) < higher_threshold)
            img(i,j) = img(i,j) + constant;
        else
            img(i,j) = constant;
        end
    end
end

figure('Name','Original Image','NumberTitle','off');
imshow(image,'Border','tight');
pause();
h1 = histogram(image);
pause();
figure('Name','Constant Output Image','NumberTitle','off');
imshow(img,'Border','tight');
pause();
h2 = histogram(img);
pause();
close all;

File: constantoutput.py
#take required libraries
import numpy as np
import cv2

image=cv2.imread("lena.bmp",0)
cv2.imshow("Original Image",image)
img=image

r = img.shape[0]
c = img.shape[1]

for i in range(0,r):
    for j in range(0,c):
        if img[i][j] > 50 & img[i][j]<150:
            img[i][j] = img[i][j] + 90
cv2.imshow("Threshold Constant",img)
